examples/alpaka/nbody/nbody.cpp

```diff
@@ -162,7 +162,7 @@ struct UpdateKernel
                     if constexpr (MappingSM == AoS)
                         return llama::mapping::AoS{arrayDims, Particle{}};
                     if constexpr (MappingSM == SoA)
-                        return llama::mapping::SoA{arrayDims, Particle{}};
+                        return llama::mapping::SoA<decltype(arrayDims), Particle, false>{arrayDims};
```

Comment:
It feels weird that we require the `<>` for SoA and AoSoA, but not for AoS.

Reply:
The story is more complicated. The angle bracket interface is actually the default, `AoS<decltype(arrayDims), Particle>{arrayDims}` is what you would write normally. Notice that the arrayDims are repeated twice, with is not great. Ideally, we could deduce the type of `arrayDims` for `AoS`. This is possible using Constructor Template Argument Deduction (CTAD). However, then we need to be able to deduce all template parameters of `AoS`. So I added an unused argument for the record dim to `AoS`, so you can write: `AoS{arrayDims, Particle{}};`, allowing the template arguments for both array and record dimensions to be deduced. However, that feature immediately fails in case we want to have more complicated constructurs.

The longer discussion is here: https://github.com/alpaka-group/llama/discussions/167. Feel free to leave a comment here!